app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3260,6 +3280,73 @@ public void disablePreloadingOfCurrentTrack() {
         loadController.disablePreloadingOfCurrentTrack();
     }
 
+    /**
+     * Set the default resolution for livestreams according to user choice in "Video and audio"
+     * settings.
+     *
+     * <p>
+     * If no stream match the minimum criteria required (video format height <= default height),
+     * then the maximum resolution limit and the maximum frame rate will be set.
+     * </p>
+     *
+     * @param videoTrackGroupArray the video {@link TrackGroupArray}, which cannot be null
+     * @param videoRendererIndex   the video renderer index
+     */
+    private void setDefaultResolutionForLivestreamsIfNeeded(
+            @NonNull final TrackGroupArray videoTrackGroupArray,
+            final int videoRendererIndex) {
+        if (isDefaultResolutionOverrideOrMaxVideoSizeAndFrameRateSet) {
+            return;
+        }
+
+        final String defaultResolution;
+        if (popupPlayerSelected()) {
+            defaultResolution = computeDefaultResolution(context,
+                    R.string.default_popup_resolution_key,
+                    R.string.default_popup_resolution_value);
+        } else {
+            defaultResolution = computeDefaultResolution(context,
+                    R.string.default_resolution_key, R.string.default_resolution_value);
+        }
+        final String[] defaultResolutionArray = defaultResolution.split("p");
+        final int defaultHeight = !defaultResolution.equals(context.getString(
+                R.string.best_resolution_key))
+                ? Integer.parseInt(defaultResolutionArray[0]) : -1;
+        final int defaultFrameRate = defaultResolutionArray.length == 2
+                ? Integer.parseInt(defaultResolutionArray[1]) : -1;
+
+        final DefaultTrackSelector.Parameters selectionParameters = trackSelector
+                .getParameters();
+        final DefaultTrackSelector.ParametersBuilder builder = selectionParameters
+                .buildUpon().clearSelectionOverrides();
+
+        for (int i = 0; i < videoTrackGroupArray.length; i++) {
+            final TrackGroup trackGroup = videoTrackGroupArray.get(i);
+            for (int j = trackGroup.length - 1; j >= 0; j--) {
+                final Format videoFormat = trackGroup.getFormat(j);
+                if ((defaultHeight == -1 || videoFormat.height <= defaultHeight)
+                        && (defaultFrameRate <= -1 || videoFormat.frameRate == defaultFrameRate)) {
+                    final DefaultTrackSelector.SelectionOverride selectionOverride =
+                            new DefaultTrackSelector.SelectionOverride(
+                                    videoRendererIndex, j);
+
+                    builder.setSelectionOverride(videoRendererIndex, videoTrackGroupArray,
+                            selectionOverride);
+                    trackSelector.setParameters(builder);
+                    isDefaultResolutionOverrideOrMaxVideoSizeAndFrameRateSet = true;
```

Comment:
This is being done independently of from where the function returns, so it should not be duplicated, but should just be put below `if (isDefaultResolutionOverrideOrMaxVideoSizeAndFrameRateSet) { return; }`